src/backends/gr.jl

```diff
@@ -1282,12 +1282,22 @@ function gr_set_window(sp, viewport_plotarea)
         gr_set_viewport_polar(viewport_plotarea)
     else
         xmin, xmax, ymin, ymax = gr_xy_axislims(sp)
+        is3d = RecipesPipeline.is3d(sp)
```

Comment:
you might want to check, if you actually wanted `has_3d_axes` here, since `is3d` is also true for e.g. contour plots, that have 3d data but 2d axes.

Reply:
I could not find a way to check if `z` axis has some data, does you have an idea for that ?

Reply:
I meant this one:
https://github.com/JuliaPlots/RecipesPipeline.jl/blob/00ed890cb63f5d65183f0a10f4172f5124a2f746/src/utils.jl#L171